crates/interpreter/src/exec.rs

```diff
@@ -195,9 +205,24 @@ impl<'m> Store<'m> {
             let mut parser = self.insts[inst_id].module.func(ptr.index());
             let mut locals = Vec::new();
             append_locals(&mut parser, &mut locals);
-            let thread = Thread::new(parser, Frame::new(inst_id, 0, &[], locals));
+            let thread = Thread::new(
+                parser,
+                Frame::new(inst_id, 0, &[], locals),
+                #[cfg(feature = "interrupt")]
+                self.interrupt,
+            );
+
+            // Disable interrupts for the start section.
```

Comment:
Indeed, my code snippet was wrong, we need to specify the store lifetime:

```rust
    fn without_interrupt<R>(&mut self, operation: impl FnOnce(&mut Store<'m>) -> R) -> R {
```

And then it can be used like that:

```rust
            let thread = Thread::new(parser, Frame::new(inst_id, 0, &[], locals));
            self.without_interrupt(|x| {
                let result = thread.run(x)?;
                assert!(matches!(result, RunResult::Done(x) if x.is_empty()));
                Ok(())
            })?;
```